modules/mod_acl_user_groups/support/acl_user_groups_checks.erl

```diff
@@ -623,6 +623,12 @@ is_owner(insert_rsc, _Context) ->
 is_owner(Id, Context) ->
     is_owner(Id, m_rsc:p_no_acl(Id, creator_id, Context), Context).
 
+is_owner(Id, undefined, #context{user_id=undefined, session_id=SessionId} = Context) ->
+    case m_rsc:p_no_acl(Id, session_owner, Context) of
+        undefined -> false;
```

Comment:
By placing the session-id as a property in the resource, you are in fact publishing it to the whole world.
Another solution could be to have the resource-id in the session process and then check against the session state.